### PR TITLE
Attempt to fix CI issues

### PR DESCRIPTION
**Pull request by adam-fowler**
This improves the stability of tests, but they still fail occasionally
Also added a test to check CMake projects work

### CODE REVIEW

File: assets/test/cmake/compile_commands.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+{
+  "directory": "/Users/adamfowler/Developer/tools/vscode-swift/assets/test/cmake",
```
Comment (0xTim) [inline assets/test/cmake/compile_commands.json:3]: Do we want these committed?

**Review by 0xTim — COMMENTED** [review]
Query about file with user paths in, otherwise LGTM

File: assets/test/cmake/compile_commands.json (hunk 1 above)
Comment (adam-fowler) [reply assets/test/cmake/compile_commands.json:3]: We need the compile_commands.json to trigger the extension for CMake projects. I'll edit it though

Comment (adam-fowler) [reply assets/test/cmake/compile_commands.json]: I've removed most of the cmake project files as they were not needed

**Comment by adam-fowler** [comment]
@swift-server-bot test this please

**Review by 0xTim — APPROVED** [review]

**Pull request merged**